refresh/index.js

```diff
@@ -48,7 +114,90 @@ module.exports = generators.Base.extend({
       }
     },
 
+    createBasicProject: function() {
+      // Check if there is a yo-rc, create one if there isn't
+      if(!this.fs.exists(this.destinationPath('.yo-rc.json'))) {
+        this.fs.writeJSON(this.destinationPath('.yo-rc.json'), {});
+      }
+
+      // Check if there is a Package.swift, create one if there isn't
+      if(!this.fs.exists(this.destinationPath('Package.swift'))) {
+        let packageSwift = helpers.generatePackageSwift(this.config);
+        this.fs.write(this.destinationPath('Package.swift'), packageSwift);
+      }
+
+      //TODO: decide what to do with the app name, e.g if the user changes the
+      //      name in the config.json file
+      // Check if there is a main.swift, create one if there isn't
+      // if(!this.fs.exists(this.destinationPath('Sources', this.config.appName, 'main.swift'))) {
+      //   this.fs.copy(this.templatePath('main.swift'),
+      //                this.destinationPath('Sources', this.config.appName, 'main.swift'));
+      // }
+
+      // Check if there is a .swiftservergenerator-project, create one if there isn't
+      if(!this.fs.exists(this.destinationPath('.swiftservergenerator-project'))) {
+        // NOTE(tunniclm): Write a zero-byte file to mark this as a valid project
+        // directory
+        this.fs.write(this.destinationPath('.swiftservergenerator-project'), '');
+      }
+
+      // Check if there is a models folder, create one if there isn't
+      if(!this.fs.exists(this.destinationPath('models', '.keep'))) {
+        this.fs.write(this.destinationPath('models', '.keep'), '');
+      }
+
+      // Check if there is a index.js, create one if there isn't
+      if (this.options.apic) {
+        if(!this.fs.exists(this.destinationPath('index.js'))) {
+          this.fs.copy(this.templatePath('apic-node-wrapper.js'),
+                       this.destinationPath('index.js'));
+        }
+      }
+    },
+
+    createModels: function() {
+      var done = this.async();
+
+      // Get the models from the generators
+      if(this.options.model) {
+        var modelList = [this.options.model];
+      }
+
+      // Check for a specification
+      if(this.spec) {
+        // Check if there are any models defined
+        if(this.spec.models) {
+          var modelList = this.spec.models;
+        } else {
+          done();
+          return;
+        }
+      }
+
```

Comment:
The code above might be simplified if the models are always passed via the spec option (ie if the property generator passed its model as a spec).